src/genericfit.jl

```diff
@@ -30,5 +30,7 @@ fit_mle(dt::Type{D}, x::AbstractArray, w::AbstractArray) where {D<:UnivariateDis
 fit_mle(dt::Type{D}, x::AbstractMatrix) where {D<:MultivariateDistribution} = fit_mle(D, suffstats(D, x))
 fit_mle(dt::Type{D}, x::AbstractMatrix, w::AbstractArray) where {D<:MultivariateDistribution} = fit_mle(D, suffstats(D, x, w))
 
+fit_mle(dt::D, args...) where {D<:Distribution} = fit_mle(typeof(dt), args...)
```

Comment:
`D` is not needed:
```suggestion
fit_mle(dist::Distribution, args...) = fit_mle(typeof(dist), args...)
```
Just adding this definition would not cause any confusion. But I also don't see a completely clear benefit of it as one could already use `typeof(dist)`.

Reply:
If `fit_mle(dist::Distribution, x)` does not exist for some distribution, this line will try `fit_mle(::Type, x)`. 
This ensures that `fit_mle(dist::Distribution, x)` will call currently implemented `fit_mle` methods that use type and not instance. 
If this is added for me, it is already something.